addons/rating/models/rating_mixin.py

```diff
@@ -79,6 +111,37 @@ def _compute_rating_stats(self):
             record.rating_count = mapping.get(record.id, {}).get('rating_count', 0)
             record.rating_avg = mapping.get(record.id, {}).get('rating_avg', 0)
 
+    def _search_rating_avg(self, operator, value):
+        if operator not in ['=', '!=', '<=', '<', '>', '>=']:
+            raise NotImplementedError('This operator %s is not supported in this search method.' % operator)
+        if not isinstance(value, (int, float)):
+            raise NotImplementedError('The value must be an integer or a float value.')
+        query = f"""
+            SELECT doc.id
+              FROM {self._table} as doc
+         LEFT JOIN rating_rating R
+                ON R.res_id = doc.id AND res_model = %s AND R.consumed = 't' AND R.rating >= %s
+          GROUP BY doc.id
+            HAVING COALESCE(avg(R.rating), 0) {operator} %s
+        """
+        return [(
+            'id',
+            'inselect' if operator != '!=' else 'not inselect',
+            (query, (self._name, RATING_LIMIT_MIN, value,))
+        )]
+
+    @api.depends('rating_avg')
+    def _compute_rating_avg_text(self):
+        for record in self:
+            if record.rating_avg >= 3.66:
+                record.rating_avg_text = 'top'
+            elif record.rating_avg >= 2.33:
```

Comment:
Would use explicit globals, like RATING_AVG_TOP, ... to make them more explicit (and possibly change them when we want more "top" projects)